### PR TITLE
Handle nil values

### PR DESCRIPTION
**Pull request by ekohl**
I'm not 100% sure this improves things, but sometimes we get error mails on on nil values and this should improve the debug options.

### CODE REVIEW

File: puppet/modules/redmine/files/git_repos.sh [hunk 1]
```diff
@@ -36,6 +36,10 @@ fi
 # Sync repositories for all known git repos
 curl -s $REDMINE_REPOS | ruby -rjson -e '
 JSON.load(STDIN).each do |project_name,repos|
+  unless repos
```
Comment (evgeni) [inline puppet/modules/redmine/files/git_repos.sh]: I think you're chasing the wrong `each`. The error I see in my INBOX is 
```
-e:2:in `<main>': undefined method `each' for nil:NilClass (NoMethodError)
```
*line 2*, but `repos.each` is on line 3 (the first line being the empty line at `ruby -rjson -e '`).

I think the failing `each` is `JSON.load(STDIN).each`. Test:

```
% cat /tmp/test.sh 
#!/bin/sh

echo "null" | ruby -rjson -e '
JSON.load(STDIN).each do |p|
	puts p
end'

% bash /tmp/test.sh
Traceback (most recent call last):
-e:2:in `<main>': undefined method `each' for nil:NilClass (NoMethodError)
```

**Comment by ekohl** [comment]
So how about this?

**Comment by evgeni** [comment]
Aren't we fetching just one url? I thought fail early helps only if you have multiple?

**Comment by ekohl** [comment]
You're right. It doesn't imply `--fail`. Updated.
>        -f, --fail
>              (HTTP)  Fail silently (no output at all) on server errors. This is mostly done to better enable scripts etc to bet‐
>              ter deal with failed attempts. In normal cases when an HTTP server fails to deliver a document, it returns an  HTML
>              document stating so (which often also describes why and more). This flag will prevent curl from outputting that and
>              return error 22.
>
>              This method is not fail-safe and there are occasions where non-successful response codes will slip  through,  espe‐
>              cially when authentication is involved (response codes 401 and 407).

**Comment by evgeni** [comment]
@ekohl I think this script needs a `set -o pipefail`, as otherwise `--fail` will fail the `curl` call, but the pipe will still be executed.

```
% curl --fail --silent https://die-welt.net/nope.json | ruby -rjson -e 'JSON.load(STDIN).each {|p| puts p}'
Traceback (most recent call last):
-e:1:in `<main>': undefined method `each' for nil:NilClass (NoMethodError)
```

**Comment by ekohl** [comment]
Updated to set pipefail.

**Pull request merged**